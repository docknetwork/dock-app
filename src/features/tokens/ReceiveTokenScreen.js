import Clipboard from '@react-native-community/clipboard';
import { Box, Stack } from 'native-base';
import React from 'react';
import { Dimensions } from 'react-native';
import QRCode from 'react-native-qrcode-svg';
import Share from 'react-native-share';
import { useSelector } from 'react-redux';
import { PolkadotIcon } from 'src/components/PolkadotIcon';
import { formatAddress } from 'src/core/format-utils';
import { showToast } from 'src/core/toast';
import {
  Button,
  Content, ScreenContainer,
  Theme,
  Typography
} from '../../design-system';
import { translate } from '../../locales';
import { accountSelectors } from '../accounts/account-slice';


export function ReceiveTokenScreen({
  address,
  accountName,
  accountIcon,
  onCopyAddress,
  onShareAddress,
}) {
  const qrSize = Dimensions.get('window').width * 0.7;

  return (
    <ScreenContainer testID="unlockWalletScreen">
      <Stack alignItems="center">
        <Typography variant="h1" mb={2}>
          {translate('receive_token.title')}
        </Typography>
        <Typography variant="screen-description">
          {translate('receive_token.description')}
        </Typography>
      </Stack>
      <Content marginLeft={26} marginRight={26}>
        <Box
          bg={Theme.colors.secondaryBackground}
          p={5}
          alignItems="center"
          borderRadius={12}
          my={7}>
          <Box>
            <QRCode
              value={address}
              size={qrSize}
              color={Theme.colors.secondaryBackground}
            />
          </Box>
          <Stack direction="row" alignItems="center" mt={3}>
            <Box pr={2}>{accountIcon}</Box>
            <Stack direction="column" alignItems="flex-start">
              <Typography variant="h3">{accountName}</Typography>
              <Typography>{formatAddress(address)}</Typography>
            </Stack>
          </Stack>
        </Box>
        <Button onPress={onCopyAddress} colorScheme="tertiary" mb={4}>
          {translate('receive_token.copy_address')}
        </Button>
        <Button onPress={onShareAddress} colorScheme="tertiary">
          {translate('receive_token.share_address')}
        </Button>
      </Content>
    </ScreenContainer>
  );
}

export function ReceiveTokenContainer({route}) {
  const {address} = route.params || {};
  const accountDetails = useSelector(accountSelectors.getAccountById(address));

  const handleCopyAddress = () => {
    Clipboard.setString(address);
    showToast({
      message: translate('receive_token.address_copied'),
    });
  };

  const handleShareAddress = () =>
    Share.open({
      message: address,
    });

  return (
    <ReceiveTokenScreen
      onCopyAddress={handleCopyAddress}
      onShareAddress={handleShareAddress}
      address={address}
      accountName={accountDetails.meta.name}
      accountIcon={<PolkadotIcon address={address} />}
    />
  );
}
