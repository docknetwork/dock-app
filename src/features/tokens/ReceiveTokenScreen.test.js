import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {ReceiveTokenScreen} from './ReceiveTokenScreen';

const mockStore = configureMockStore();

describe('ReceiveTokensScreen', () => {
  it('should render correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(<ReceiveTokenScreen address={'3B9W5fS4ygoUVnGhzTk7sihSdhCyJQRNMQ7aJfDdsfWb16f1'} />, {
      context: {store: mockStore(initialState)},
    });
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
