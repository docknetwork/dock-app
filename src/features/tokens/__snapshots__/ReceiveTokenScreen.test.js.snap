// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReceiveTokensScreen should render correctly 1`] = `
<StyledNativeComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "inlineStyle": InlineStyle {
        "rules": Array [
          "
  background-color: ",
          "#18181B",
          ";
  flex: 1;
",
        ],
      },
      "render": [Function],
      "shouldForwardProp": undefined,
      "styledComponentId": "StyledNativeComponent",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  testID="unlockWalletScreen"
>
  <Memo(ForwardRef(Stack))
    alignItems="center"
  >
    <Styled(Component)
      mb={2}
      variant="h1"
    >
      Send DOCK
    </Styled(Component)>
    <Styled(Component)
      variant="screen-description"
    >
      Share your address
    </Styled(Component)>
  </Memo(ForwardRef(Stack))>
  <Styled(ScrollView)
    marginLeft={26}
    marginRight={26}
  >
    <Memo(ForwardRef(Box))
      alignItems="center"
      bg="#27272A"
      borderRadius={12}
      my={7}
      p={5}
    >
      <Memo(ForwardRef(Box))>
        <QRCode
          color="#27272A"
          size={525}
          value="3B9W5fS4ygoUVnGhzTk7sihSdhCyJQRNMQ7aJfDdsfWb16f1"
        />
      </Memo(ForwardRef(Box))>
      <Memo(ForwardRef(Stack))
        alignItems="center"
        direction="row"
        mt={3}
      >
        <Memo(ForwardRef(Box))
          pr={2}
        />
        <Memo(ForwardRef(Stack))
          alignItems="flex-start"
          direction="column"
        >
          <Styled(Component)
            variant="h3"
          />
          <Styled(Component)>
            3B9W5fS4y...DdsfWb16f1
          </Styled(Component)>
        </Memo(ForwardRef(Stack))>
      </Memo(ForwardRef(Stack))>
    </Memo(ForwardRef(Box))>
    <Button
      colorScheme="tertiary"
      mb={4}
    >
      Copy address
    </Button>
    <Button
      colorScheme="tertiary"
    >
      Share address
    </Button>
  </Styled(ScrollView)>
</StyledNativeComponent>
`;
